metadataserver/iam.go

```diff
@@ -75,9 +75,19 @@ func newIamProxy(ctx context.Context, router *mux.Router, config types.MetadataS
 	s := session.Must(session.NewSession())
 
 	if config.Region != "" {
+		endpoint := fmt.Sprintf("sts.%s.amazonaws.com", config.Region)
+		if config.STSEndpoint != "" {
+			endpoint = config.STSEndpoint
+		}
+
 		stsAwsCfg := aws.NewConfig().
 			WithRegion(config.Region).
-			WithEndpoint(fmt.Sprintf("sts.%s.amazonaws.com", config.Region))
+			WithEndpoint(endpoint)
+
+		if config.DisableSTSEndpointSSL {
```

Comment:
Why would you ever want this?

Reply:
For testing - otherwise STS fails to connect to the mock endpoints.  I suppose I could put it behind a `STSTestMode` flag or something.

Reply:
Can you use the Golang httptest TLS functionality? https://golang.org/pkg/net/http/httptest/#NewTLSServer